homeassistant/components/sonos/speaker.py

```diff
@@ -200,6 +201,7 @@ async def async_handle_new_entity(self, entity_type: str) -> None:
         if self._platforms_ready == PLATFORMS:
             await self.async_subscribe()
             self._is_ready = True
+        self._resubscription_lock = asyncio.Lock()
```

Comment:
Why do we create a new lock here? Why not create the lock in init method of this class?

Reply:
It must be created from within an async context. All `SonosSpeaker` instances are created in a sync callback now.

Reply:
Shouldn't we create the lock before we subscribe?

Reply:
Yes, sorry. I misremembered the purpose of this method.